### DISCUSSION TITLE
Adding mdformat & mdit-py-plugins maintainers

### DISCUSSION THREAD

**Discussion by KyleKing** [executablebooks/meta#1196] [General] [answered]
Hi! If you would be open to adding maintainers for mdformat-related projects, I would like to volunteer. While I have limited time, I can help with issue triage, limited PR approval, and software maintenance.

I've released a few plugins (`mdformat-mkdocs`, `mdformat-obsidian`, `mdformat-gfm-alerts`, `mdformat-plugin-template`, etc.) and have made a few very minor contributions to `mdformat` and `mdit-py-plugins`. This year, I was granted minimal rights to `mdformat-gfm` (https://github.com/hukkin/mdformat-gfm/issues/31#issuecomment-1916763755) where I believe I've made a positive impact and is an example of how I could help. I would like to do more for a project I depend on.

I know that [EBP](https://compass.executablebooks.org/en/latest/governance.html)/[MyST](https://compass.executablebooks.org/en/latest/development/conventions.html#structure-of-ebp-repositories) have been active, but the markdown-it-side seems to have seen less activity (#841, #946, #961, etc.).

Who could make a decision about adding maintainers? In the past, I've interacted on Issues and PRs with @hukkin and @chrisjsewell

**Comment by agoose77** [comment, marked as answer]
@KyleKing this is a generous offer, it's much appreciated by the wider EB team. We are always looking for new maintainers! Your expertise in this space will be highly valued.

@hukkin is one of the main contributors to our `mdformat` related repos, so it would be nice to have his sign off on this. If we don't hear anything within the next two weeks, then I suppose we can move on unilaterally (it may be that Taneli is not available). I'll also ping @choldgraf to get a sign-off as this would be the first time that I've added a new contributor.

If you don't hear anything in the next week (the EB team are generally quite busy at the moment), please re-ping me!

**Reply by KyleKing** [reply]
Thanks, I am happy to help!

It looks like @hukkin may not be available right now, so I don't mind waiting if needed or if there is someone else who could weigh in

_Edit_: once I have permission, there are a couple of trivial PRs that should get merged and released linked from here: https://github.com/KyleKing/mdformat-admon/issues/15

**Reply by choldgraf** [reply]
This sounds great. Please don't ask me for permission to add other people - you all have much better information than I do to make these decisions

**Reply by KyleKing** [reply]
@agoose77, what are the next steps?

**Reply by agoose77** [reply]
Let me get to this on Monday!

**Reply by KyleKing** [reply]
Just following up. Would you have time today?

The mdformat projects follow these community guidelines and technical process, right? https://compass.executablebooks.org/en/latest/development/conventions.html

**Reply by agoose77** [reply]
@KyleKing thanks for being patient -- it's been a bit busier than I had expected and I lost sight of this. We are thrilled to have more maintainers; I'll work on setting you up.

**Reply by KyleKing** [reply]
Thanks! Let me know if there is anything I need to know. I'll slowly start triaging open Issues and PR over the next month to clear the backlog

**Reply by agoose77** [reply]
This is great news. Thanks for bearing with us; there's so much to do at the moment that it can take a while to unblock things like adding maintainers! Hopefully you're not blocked now :)

**Reply by KyleKing** [reply]
@agoose77 / @hukkin, who has access to the Github Secrets for `mdformat-myst`? I believe the `PYPI_TOKEN` might be missing or no longer active blocking a new release: https://github.com/executablebooks/mdformat-myst/actions/runs/10533867668/job/29190590877

**Reply by KyleKing** [reply]
Separately, could I be given similar maintainer access to ~~https://github.com/executablebooks/mdit-py-plugins~~ (**_Correction_**: https://github.com/executablebooks/mdformat)? There are documentation PRs that should get merged: https://github.com/executablebooks/mdformat/pull/427#issuecomment-2178905963

**Reply by chrisjsewell** [reply]
> Separately, could I be given similar maintainer access to https://github.com/executablebooks/mdit-py-plugins? There are documentation PRs that should get merged: https://github.com/executablebooks/mdformat/pull/427#issuecomment-2178905963

Heya, you mention mdit-py-plugins here, but then you link to documentation PRs of mdformat?
There are no outstanding PRs on mdit-py-plugins ready for merge

**Reply by chrisjsewell** [reply]
Ah I think you meant https://github.com/executablebooks/mdformat-plugin yeh? Sounds good to me, thanks for the work 😄

**Reply by KyleKing** [reply]
Haha yeah, too many tabs open. I fixed the link above for https://github.com/executablebooks/mdformat to merge the documentation PRs

On `mdit-py-plugins`, there are some issues that could potentially be closed or some minor fixes that have been requested that would be trivial to patch (trivial fixes: https://github.com/executablebooks/mdit-py-plugins/issues/92 and https://github.com/executablebooks/mdit-py-plugins/issues/89 and https://github.com/executablebooks/mdit-py-plugins/issues/97#issuecomment-2113468594, https://github.com/executablebooks/mdit-py-plugins/issues/44, https://github.com/executablebooks/mdit-py-plugins/issues/60, https://github.com/executablebooks/mdit-py-plugins/issues/62, and issues that can be closed: https://github.com/executablebooks/mdit-py-plugins/issues/39, https://github.com/executablebooks/mdit-py-plugins/issues/96)

Could I be given similar minimal maintenance permission to the other mdit repos?

- https://github.com/executablebooks/mdit-py-plugins
- https://github.com/executablebooks/mdurl
- https://github.com/executablebooks/markdown-it-plugin-template
- https://github.com/executablebooks/markdown-it-docutils
- https://github.com/executablebooks/markdown-it-amsmath
- https://github.com/executablebooks/markdown-it-dollarmath
- https://github.com/executablebooks/markdown-it-myst-extras

**Reply by KyleKing** [reply]
Just following up, @agoose77, who can has admin access to PyPi and the Github repo Action secrets? There may be other projects, but at least mdformat-myst can't be released without updating the PyPi token: https://github.com/executablebooks/mdformat-myst/issues/30

**Comment by mattkang** [comment]
@agoose77 @KyleKing Any update on this? Would love to see this PR merged https://github.com/executablebooks/mdformat-myst/pull/28, and it looks like the existing maintainers may not be available

**Comment by KyleKing** [comment]
@agoose77 would you be able to grant me access to `mdformat-footnote` or add the secret `PYPI_TOKEN`?

I want to publish changes to address the recent mdformat v1 release (https://github.com/executablebooks/mdformat-footnote/pull/14), but I'm blocked by the lack of credentials

If I have organization access on PyPi, I can also setup the more secure publishing mechanism recommended by PyPi and started here: https://github.com/executablebooks/mdformat-footnote/pull/9 (cc @gaige)

**Reply by gaige** [reply]
We shouldn't be publishing with tokens any longer, that's what the OIDC publisher is intended for.  Thanks for raising this issue. I'm in the process of taking a look at it. Clearly I got half way through the update this summer and didn't realize I had a typo in there (a bit of copy-paste from another repo I was updating).

**Reply by KyleKing** [reply]
Are the other `executablebook` PyPi projects configured for Trusted Publishing as well and could have their actions updated? Or only the footnote one so far?

There was a failure in the mdformat-footnote upload/signing after publishing, so I submitted a change that should keep the actions up to date and resolve the error for future releases: https://github.com/executablebooks/mdformat-footnote/pull/16

**Reply by gaige** [reply]
I don't know how it's being done across the rest of ExecutableBooks.  I think I only have access to the PyPI repos that I brought in.
